venues/ICLR.cc/2018/Conference/process/officialReviewProcess_withRevision.js

```diff
@@ -0,0 +1,61 @@
+//EDIT ME
+
+function(){
+    var or3client = lib.or3client;
+
+    var origNote = or3client.or3request(or3client.notesUrl+'?id='+note.forum, {}, 'GET', token);
+    var list = note.invitation.replace(/_/g,' ').split('/');
+    list.splice(list.indexOf('-',1));
+    var conference = list.join(' ');
+
+    origNote.then(function(result) {
+      var forum = result.notes[0];
+      var note_number = forum.number;
+
+      var reviewers = ['ICLR.cc/2018/Conference/Paper' + note_number + '/Reviewers'];
+      var areachairs = ['ICLR.cc/2018/Conference/Paper' + note_number + '/Area_Chair'];
+      var authors = forum.content.authorids;
+
+      var areachair_mail = {
+        "groups": areachairs,
+        "subject": "Review posted to your assigned paper: \"" + forum.content.title + "\"",
+        "message": "A submission to " + conference + ", for which you are an official area chair, has received an official review. \n\nTitle: " + note.content.title + "\n\nComment: " + note.content.review + "\n\nTo view the review, click here: " + baseUrl + "/forum?id=" + note.forum
+      };
+      var areachairMailP = or3client.or3request( or3client.mailUrl, areachair_mail, 'POST', token );
+
+      return areachairMailP;
+    })
+    .then(result => or3client.addInvitationNoninvitee(note.invitation, note.signatures[0], token))
+    .then(result => or3client.or3request(or3client.inviteUrl+'?id='+note.invitation, {}, 'GET', token))
+    .then(result => {
+      var reviewInvitation = result.invitations[0];
+      var signatureComponents = note.signatures[0].split('/');
+      console.log('signature components: ' + signatureComponents);
+      var anonReviewerPaper = signatureComponents
+      var revisionInvitation = {
+        'id': 'ICLR.cc/2018/Conference/-/'+signatureComponents[4]+'/'+signatureComponents[3]+'/Revise_Review',
```

Comment:
We could create all the revise_review invitations in the review_revisions, so we don't need to change this process function.

Reply:
I think we need to create them in the process function, because the revision invitation needs to set its referent field to the ID of the review note, which won't exist until the review is posted

Reply:
You are right!